java_src/foam/dao/index/ValueIndex.java

```diff
@@ -7,12 +7,12 @@
 
 import foam.core.FObject;
 import foam.dao.Sink;
+import foam.mlang.order.Comparator;
 import foam.mlang.predicate.Predicate;
-import java.util.Comparator;
 
 public class ValueIndex implements Index {
   
-  protected static ValueIndex instance_ = null;
+  protected static ValueIndex instance_ = new ValueIndex();
   
   protected FObject value;
```

Comment:
ValueIndex should not have a value. That's stored in the state which is passed it.